Minor: remove duplicate docs extra and clean up comments

harshil-sanghvi opened the pull request:
Minor: remove duplicate docs extra and clean up comments

- pyproject.toml: drop duplicate sphinx-design entry in docs extras (keep >=0.6.0)
- sklearn/tree/_classes.py: grammar fix in comments ("nan is only supported")
- sklearn/utils/fixes.py: standardize "SciPy" capitalization in TODOs

<!--
🙌 Thanks for contributing a pull request!

👀 Please ensure you have taken a look at the contribution guidelines:
https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md

✅ In particular following the pull request checklist will increase the likelihood
of having maintainers review your PR:
https://scikit-learn.org/dev/developers/contributing.html#pull-request-checklist

📋 If your PR is likely to affect users, you will need to add a changelog entry
describing your PR changes, see:
https://github.com/scikit-learn/scikit-learn/blob/main/doc/whats_new/upcoming_changes/README.md
-->

#### Reference Issues/PRs
N/A

#### What does this implement/fix? Explain your changes.
- Removes a duplicate `sphinx-design` entry in `[project.optional-dependencies].docs` (retains newer version spec).
- Fixes minor comment grammar in `sklearn/tree/_classes.py`.
- Standardizes "SciPy" capitalization in TODO comments in `sklearn/utils/fixes.py`.
- No behavior changes; docs/dependency list cleanliness only.

#### Any other comments?
- Label suggestion: "No Changelog Needed".
- Verified no linter issues locally.

harshil-sanghvi committed:
Minor maintenance: remove duplicate docs extra and fix small comment nits\n\n- pyproject.toml: drop duplicate sphinx-design entry in docs extras\n- sklearn/tree/_classes.py: grammar: 'nan is only supported'\n- sklearn/utils/fixes.py: standardize 'SciPy' capitalization in TODOs

harshil-sanghvi committed:
Fix ruff E501: wrap long comment lines in tree/_classes.py

virchan commented on sklearn/tree/_classes.py:
```diff
@@ -1099,7 +1099,8 @@ def predict_log_proba(self, X):
         tags = super().__sklearn_tags__()
-        # XXX: nan is only support for dense arrays, but we set this for common test to
+        # XXX: nan is only supported for dense arrays, but we set this for
+        # common test to
```
Please merge the two lines so that “to pass” stays on the same line as the preceding text.

virchan commented on sklearn/tree/_classes.py:
```diff
@@ -1442,7 +1443,8 @@ def _compute_partial_dependence_recursion(self, grid, target_features):
         tags = super().__sklearn_tags__()
-        # XXX: nan is only support for dense arrays, but we set this for common test to
+        # XXX: nan is only supported for dense arrays, but we set this for
+        # common test to
```
Please refer to my previous comment.

harshil-sanghvi replied:
done

virchan reviewed (commented):
Thanks for the PR, @harshil-sanghvi.

I have a few comments.

harshil-sanghvi committed:
Fix comment formatting in DecisionTree classes

- Consolidate line breaks in comments for better readability
- Remove unnecessary line breaks in __sklearn_tags__ method comments

virchan reviewed (approved):
Thanks, @harshil-sanghvi.

jeremiedbb reviewed (approved):
Thanks @harshil-sanghvi
